x/blob/ante/ante.go

```diff
@@ -33,8 +36,12 @@ func (d MinGasPFBDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool
 		// NOTE: here we assume only one PFB per transaction
 		if pfb, ok := m.(*types.MsgPayForBlobs); ok {
 			if gasPerByte == 0 {
-				// lazily fetch the gas per byte param
-				gasPerByte = d.k.GasPerBlobByte(ctx)
+				if ctx.BlockHeader().Version.App <= v2.Version {
+					// lazily fetch the gas per byte param
+					gasPerByte = d.k.GasPerBlobByte(ctx)
+				} else {
+					gasPerByte = appconsts.GasPerBlobByte(v3.Version)
```

Comment:
I think it'll still work in v4 because no matter what version you pass into the function it will return v3.GasPerBlobByte  which Is slightly unintuitive. I propose we remove the option to pass arguments unless we actually use them.

Reply:
Yes but that's incorrect behaviour. It should be

```suggestion
					gasPerByte = appconsts.GasPerBlobByte(ctx.BlockHeader().Version.App)
```